components/brave_today/browser/direct_feed_controller.cc

```diff
@@ -211,7 +230,7 @@ void DirectFeedController::FindFeeds(
 
             // Response is valid, but still might not be a feed
             ParseFeedDataOffMainThread(
-                feed_url, body_content,
+                feed_url, charset, std::move(body_content),
```

Comment:
This `std::move` of `body_content` is problematic since it means `body_content` is empty when accessed later in the callback if the content was HTML and not RSS. I'm pretty sure it's causing https://github.com/brave/brave-browser/issues/26550